app/Http/Controllers/Api/Places/SpecialFeaturesController.php

```diff
@@ -0,0 +1,173 @@
+    ReadSpecialFeatureAction,
+    CollectionSpecialFeatureAction
+};
+use Hedonist\Actions\Place\SpecialFeature\{
+    CreateSpecialFeatureRequest,
+    DeleteSpecialFeatureRequest,
+    ReadSpecialFeatureRequest,
+    CollectionSpecialFeatureRequest
+};
+use Hedonist\Actions\Place\SpecialFeature\{
+    CreateSpecialFeatureResponse,
+    DeleteSpecialFeatureResponse,
+    ReadSpecialFeatureResponse,
+    CollectionSpecialFeatureResponse
+};
+
+class SpecialFeaturesController extends ApiController
+{
+    /** @var CreateSpecialFeatureAction **/
+    private $createSpecialFeatureAction;
+    /** @var DeleteSpecialFeatureAction **/
+    private $deleteSpecialFeatureAction;
+    /** @var ReadSpecialFeatureAction **/
+    private $readSpecialFeatureAction;
+    /** @var CollectionSpecialFeatureAction **/
+    private $collectionSpecialFeatureAction;
+
+    /**
+     * SpecialFeaturesController constructor.
+     * @param CreateSpecialFeatureAction $createSpecialFeatureAction
+     * @param DeleteSpecialFeatureAction $deleteSpecialFeatureAction
+     * @param ReadSpecialFeatureAction $readSpecialFeatureAction
+     * @param CollectionSpecialFeatureAction $collectionSpecialFeatureAction
+     */
+    public function __construct(
+        CreateSpecialFeatureAction $createSpecialFeatureAction,
+        DeleteSpecialFeatureAction $deleteSpecialFeatureAction,
+        ReadSpecialFeatureAction $readSpecialFeatureAction,
+        CollectionSpecialFeatureAction $collectionSpecialFeatureAction
+    )
+    {
+        $this->createSpecialFeatureAction = $createSpecialFeatureAction;
+        $this->deleteSpecialFeatureAction = $deleteSpecialFeatureAction;
+        $this->readSpecialFeatureAction = $readSpecialFeatureAction;
+        $this->collectionSpecialFeatureAction = $collectionSpecialFeatureAction;
+    }
+
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function index()
+    {
+        try {
+            /** @var CollectionSpecialFeatureResponse $response */
+            $response = $this->collectionSpecialFeatureAction->execute(
+                new CollectionSpecialFeatureRequest()
+            );
+        } catch (\Exception $ex) {
+            // base api controller
+            return $this->errorResponse($ex->getMessage(), 400);
+        }
+
+        $responseArray = [];
+        foreach($response->getCollection() as $item){
+            $id = $item->getId();
+            $responseArray[$id] = [
+                'id' => $id,
+                'name' => $item->getName()
+            ];
+        }
+
+        // base api controller
+        return $this->successResponse($responseArray, 201);
+    }
+
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $httpRequest
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function store(CreateSpecialFeatureHttpRequest $httpRequest)
```

Comment:
names of methods of controller must be in the imperative notation (e.g. storeSpecialFeature,or createSpecialFeature)